### ISSUE TITLE
enhance configuration of environment variables

### ISSUE THREAD

**Issue by tomeichlersmith** [tomeichlersmith/denv#48]
In order to help isolate the denv from the host environment, it will be helpful to allow the user to be specific about which variables are shared between the host and the denv. We could also have a config mode that enables the current behavior: sharing all environment variables (except special and weird ones).

I'm thinking we'd introduce three new variables to the config.

- `env_var_copy_all`: `1` if doing the current behavior and `0` otherwise
- `env_var_copy`: space-separated string of environment variables to copy into the denv, special variables will throw an error (e.g. `HOME` cannot be in this list) and weird values will throw an error (e.g. can't copy in variables with newlines), ignored if `env_var_copy_all` is set to `1`
    - basically, this is a replacement for the `printenv | ...` stuff that is currently being used to deduce the environment
- `env_var_set`: space-separated `key=value` pairs to define for within the denv, allows for the user to set a variable for in the denv that may have a different variable on the host

**Comment by tomeichlersmith** [comment]
https://github.com/tomeichlersmith/denv/blob/facca21217852cb9de2abf33010755d685a0a39e/denv#L231-L232
